src/components/Personalization/groupDecisions.js

```diff
@@ -57,50 +62,68 @@ const splitDecisions = (decisions, schema) => {
   return { matchedDecisions, unmatchedDecisions };
 };
 
-const extractDecisionsByScope = (decisions, scope) => {
+const appendScopeDecision = (scopeDecisions, decision) => {
+  if (!scopeDecisions[decision.scope]) {
+    scopeDecisions[decision.scope] = [];
+  }
+  scopeDecisions[decision.scope].push(decision);
+};
+
+const isViewScope = scopeDetails =>
+  scopeDetails.characteristics &&
+  scopeDetails.characteristics.scopeType &&
+  scopeDetails.characteristics.scopeType === VIEW_SCOPE_TYPE;
+
+const extractDecisionsByScope = decisions => {
   const pageWideScopeDecisions = [];
-  const nonPageWideScopeDecisions = {};
+  const nonPageWideScopeDecisions = [];
+  const viewScopeDecisions = {};
 
   if (isNonEmptyArray(decisions)) {
     decisions.forEach(decision => {
-      if (decision.scope === scope) {
+      if (decision.scope === PAGE_WIDE_SCOPE) {
         pageWideScopeDecisions.push(decision);
+      } else if (isViewScope(decision.scopeDetails)) {
+        appendScopeDecision(viewScopeDecisions, decision);
       } else {
-        if (!nonPageWideScopeDecisions[decision.scope]) {
-          nonPageWideScopeDecisions[decision.scope] = [];
-        }
-        nonPageWideScopeDecisions[decision.scope].push(decision);
+        nonPageWideScopeDecisions.push(decision);
       }
     });
   }
 
-  return { pageWideScopeDecisions, nonPageWideScopeDecisions };
+  return {
+    pageWideScopeDecisions,
+    nonPageWideScopeDecisions,
+    viewScopeDecisions
+  };
 };
 
 const groupDecisions = unprocessedDecisions => {
   const decisionsGroupedByRedirectItemSchema = splitDecisions(
     unprocessedDecisions,
     REDIRECT_ITEM
   );
-  const decisionsGroupedByDomActionSchema = splitDecisions(
+  const decisionsGroupedByRenderableSchemas = splitDecisions(
     decisionsGroupedByRedirectItemSchema.unmatchedDecisions,
-    DOM_ACTION
+    DOM_ACTION,
+    DEFAULT_CONTENT_ITEM
   );
 
   const {
     pageWideScopeDecisions,
-    nonPageWideScopeDecisions
+    nonPageWideScopeDecisions,
+    viewScopeDecisions
   } = extractDecisionsByScope(
-    decisionsGroupedByDomActionSchema.matchedDecisions,
-    PAGE_WIDE_SCOPE
+    decisionsGroupedByRenderableSchemas.matchedDecisions
   );
 
   return {
     redirectDecisions: decisionsGroupedByRedirectItemSchema.matchedDecisions,
     pageWideScopeDecisions,
-    viewDecisions: nonPageWideScopeDecisions,
-    nonAutoRenderableDecisions:
-      decisionsGroupedByDomActionSchema.unmatchedDecisions
+    viewDecisions: viewScopeDecisions,
+    nonAutoRenderableDecisions: decisionsGroupedByRenderableSchemas.unmatchedDecisions.concat(
+      nonPageWideScopeDecisions
```

Comment:
I would suggest to keep the old order of extracting the decisions:
1. Extract the `redirect` decisions by Redirect Schema(no changes needed).
2. Extract from remaining decisions from step 1 the `renderable` decisions (by DOM_ACTION, DEFAULT_CONTENT_ITEM schema and `scopeType: view`).
3. Extract `page-wide scope` decisions and `viewDecisions` from the `renderableDecisions` from step 2, by `__view__` scope.
4. Remaining decisions from step 2 are the form-based composed decisions.

Otherwise, it becomes extremely hard to understand why we group then concat at the end the decisions, then also maintain the code.